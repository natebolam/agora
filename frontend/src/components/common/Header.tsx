import React, { FunctionComponent, ReactElement, useState } from "react";
import cx from "classnames";
import styles from "~/styles/components/common/Header.scss";
import { useTranslation } from "react-i18next";
import { Link } from "react-navi";
import MenuIcon from "~/assets/svg/MenuIcon";
import CloseIcon from "~/assets/svg/CloseIcon";
import Logo from "~/assets/svg/Logo";

interface HeaderMenuTypes {
  isOpen: boolean;
  onClose: () => void;
}

const HeaderMenu: FunctionComponent<HeaderMenuTypes> = ({
  isOpen,
  onClose,
}): ReactElement => {
  const body = document.getElementsByTagName("body").item(0);
  if (body) {
    body.setAttribute(
      "style",
      isOpen ? "position: fixed" : "position: initial"
    );
  }
  const { t } = useTranslation();
  return (
    <div
      className={cx(styles.header__menu, {
        [styles.header__menu_open]: isOpen,
      })}
    >
      <div className={styles.header__menu__closeButton} onClick={onClose}>
        <CloseIcon />
      </div>
      <div className={styles.header__logo}>
        <Link href="/">
          <Logo />
          {t("header.logoCaption")}
        </Link>
      </div>
      <div className={styles.header__menu__links}>
        <a href={t("tezosLinks.tezosWikiLink")}>{t("header.wikiLink")}</a>
        <a href={t("tezosLinks.getStartedLink")}>
          {t("header.getStartedLink")}
        </a>
        <Link href={t("tezosLinks.tezosGovernanceLink")}>
          {t("header.governanceLink")}
        </Link>
      </div>
    </div>
  );
};

interface Props {
  className?: string;
}

const AgoraHeader: FunctionComponent<Props> = ({ className }): ReactElement => {
  const { t } = useTranslation();
  const [isMenuOpen, setMenuOpen] = useState(false);
  return (
    <div className={cx(className, styles.header)}>
      <div className={styles.header__logo}>
        <Link href="/stage">
          <Logo />
          {t("header.logoCaption")}
        </Link>
      </div>
      <div className={styles.header__links}>
        <a href={t("tezosLinks.tezosWikiLink")}>{t("header.wikiLink")}</a>
        <a href={t("tezosLinks.getStartedLink")}>
          {t("header.getStartedLink")}
        </a>
        <Link href={t("tezosLinks.tezosGovernanceLink")}>
          {t("header.governanceLink")}
        </Link>
      </div>
      <div
        className={styles.header__expandMenuButton}
        onClick={(): void => setMenuOpen(true)}
      >
        <MenuIcon />
      </div>
      <div>
        <HeaderMenu
          isOpen={isMenuOpen}
          onClose={(): void => setMenuOpen(false)}
        />
      </div>
    </div>
  );
};

export default AgoraHeader;
