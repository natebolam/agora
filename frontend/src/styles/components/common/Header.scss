.header {
  display: flex;
  font-family: 'Inter', sans-serif;
  color: #0e151a;
  margin: 0.5rem 0 2rem;
  padding: 0 0.5rem 0.5rem;
  border-bottom: 1px solid #d1dae0;

  > div {
    display: flex;

    > * {
      display: flex;
      align-items: center;
    }
  }
}

.header__right {
  margin-left: auto;

  a {
    color: #0e151a;
    text-decoration: none;
    font-size: 15px;
    font-weight: 600;
    padding: 0 0.5rem;
    margin: 0 0.5rem;
  }
}

.header__logo {
  font-size: 24px;
  font-weight: 800;
  letter-spacing: 0.3em;
  height: 100%;
  margin-left: 1rem;
  color: #0e151a;
  text-decoration: none;
}
