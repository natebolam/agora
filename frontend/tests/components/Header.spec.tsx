import * as React from "react";
import * as renderer from "react-test-renderer";
import "../testUtils/setupTest";
import Header from "../../src/components/common/Header";
import { StaticRouter } from "react-router";

describe("<Header />", (): void => {
  it("renders", (): void => {
    const tree = renderer
      .create(
        <StaticRouter>
          <Header />
        </StaticRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
