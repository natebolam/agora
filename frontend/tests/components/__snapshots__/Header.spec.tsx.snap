// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<div>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      AGORA
    </a>
  </div>
  <div>
    <a
      href="#"
    >
      Tezos Wiki
    </a>
    <a
      href="#"
    >
      Get Started
    </a>
    <a
      href="#"
    >
      Tezos Governance
    </a>
  </div>
</div>
`;
